packages/eslint-plugin/docs/rules/no-misused-promises.mdx

```diff
@@ -102,6 +102,7 @@ The following options are supported:
 - `attributes`: Disables checking an asynchronous function passed as a JSX attribute expected to be a function that returns `void`
 - `properties`: Disables checking an asynchronous function passed as an object property expected to be a function that returns `void`
 - `returns`: Disables checking an asynchronous function returned in a function whose return type is a function that returns `void`
+- `subtypes`: Disables checking an asynchronous method in a type whose extended/implemented heritage type expects that method to return `void`
```

Comment:
At first called this `methods` but figured `subtypes` might be a bit more specific to the case we're handling. But maybe there's a better name we could use for this